### DISCUSSION TITLE
TSDF fusion based on NGP rendered color and depth

### DISCUSSION THREAD

**Discussion by liruiw** [NVlabs/instant-ngp#684] [General]
Hello,

I am looking for ways to do RGB-D fusion based on the camera trajectory. In addition to rendered [RGB](https://github.com/NVlabs/instant-ngp/blob/master/scripts/run.py#L257) image, I have also added depth output. However, when I use the camera poses in transforms.json along with the RGB-D information, the fused TSDF does not seem correct. I think I might be missing a pose transforms based on NGP conventions. 

I am also curious if there is a way to get metric-scale depth and reconstruction, given known camera poses and potential depth supervision. Thanks!

**Comment by KayChou** [comment]
I am doing the same thing now. I want to ask that how to save the depth images to local. I notice that when I use GUI and set render_mode to 'depth', I can see the depth images in the window. Then what can i do to save the depth images to local?

**Reply by felixgeen** [reply]
I would also like to know about how to export the depth pass using the base_cam.json path
